lib/Doctrine/Persistence/Mapping/Driver/ColocatedMappingDriver.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+namespace Doctrine\Persistence\Mapping\Driver;
+
+use Doctrine\Persistence\Mapping\MappingException;
+use FilesystemIterator;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use RecursiveRegexIterator;
+use ReflectionClass;
+use RegexIterator;
+
+use function array_merge;
+use function array_unique;
+use function assert;
+use function get_declared_classes;
+use function in_array;
+use function is_dir;
+use function preg_match;
+use function preg_quote;
+use function realpath;
+use function str_replace;
+use function strpos;
+
+/**
+ * The ColocatedMappingDriver reads the mapping metadata located near the code.
+ */
+trait ColocatedMappingDriver
```

Comment:
why trait instead of an abstract class?

Reply:
There's a resolved comment about that in this PR.

Reply:
Ah thanks, haven't seen that :)